src/config.rs

```diff
@@ -43,14 +49,34 @@ pub struct Config {
 
     #[get = "pub"]
     #[clap(
-        default_value("/var/run/cri/cri.sock"),
+        default_value(&DEFAULT_SOCK_PATH),
         env("CRI_SOCK_PATH"),
         long("sock-path")
     )]
     /// The path to the unix socket for the server
     sock_path: PathBuf,
 }
 
+impl Config {
+    /// Return the default socket path depending if running as root or not.
+    fn default_sock_path() -> PathBuf {
+        Self::default_run_path(unistd::getuid())
+            .join(crate_name!())
+            .with_extension("sock")
+    }
+
+    /// Return the default run path depending on the provided user ID.
+    fn default_run_path(uid: Uid) -> PathBuf {
+        if uid.is_root() {
+            PathBuf::from("/var/run/").join(crate_name!())
+        } else {
+            PathBuf::from("/var/run/user")
+                .join(uid.to_string())
+                .join(crate_name!())
```

Comment:
@mrunalp do we wanna use something persistent here? I guess most distributions mount it as tmpfs, not sure if this is harmful…

Reply:
`tmpfs` is fine.